src/model/events-model.js

```diff
@@ -0,0 +1,44 @@
+import { getAllDestinations } from '../mock/destinations.js';
+import { getRandomEvent } from '../mock/events.js';
+import { getAllOffers } from '../mock/offers.js';
+import { EVENT_COUNT } from '../const.js';
+
+
+export default class EventsModel {
+  events = Array.from({ length: EVENT_COUNT }, getRandomEvent);
+
+  getAllDestinations() {
+    return getAllDestinations();
+  }
+
+  getDestinationById(id) {
+    return getAllDestinations().find((item) => item.id === id);
+  }
+
+  getOffersByType(type) {
+    return getAllOffers().find((item) => item.type === type);
+  }
+
+  getOfferById(type, id) {
+    return getAllOffers()
+      .find((item) => item.type === type)
+      .offers.find((item) => item.id === id);
+  }
+
+  getEvents() {
+    return this.events;
+  }
+
+  getEventsInfo() {
+    const eventsInfo = new Map();
+    this.events.forEach((item) => {
+      eventsInfo.set(item, {
+        destination: this.getDestinationById(item.destination),
+        selectedOffers: item.offers.map((offer) =>
+          this.getOfferById(item.type, offer)
+        ),
+      });
+    });
+    return eventsInfo;
```

Comment:
Это можно сохранить один раз

```suggestion
    const eventsInfo = new Map();
    this.events.forEach((item) => {
      eventsInfo.set(item, {
        destination: this.getDestinationById(item.destination),
        selectedOffers: item.offers.map((offer) =>
          this.getOfferById(item.type, offer)
        ),
      });
    });
    return eventsInfo;
```

Reply:
Насколько я понял, надо разделить вот так

```
  getEventsInfo(event) {
    this.eventsInfo.set(event, {
      destination: this.getDestinationById(event.destination),
      selectedOffers: event.offers.map((offer) =>
        this.getOfferById(event.type, offer)
      ),
    });
  }

  applyEventsInfo() {
    this.events.forEach((item) => {
      this.getEventsInfo(item);
    });
    return this.eventsInfo;
  }
```

Reply:
Не совсем. Если ты хочешь что-то разделить, то можно `eventsInfo` вынести в свойства модели. И отдельным методом заполнять этот `Map`.
И в презентере эту часть изолировать в отдельном методе и просто вызвать в init()
```
this.tripEvents = [...this.eventsModel.getEvents()];
    this.tripEventsInfo = new Map([...this.eventsModel.getEventsInfo()]);
```